generic.go

```diff
@@ -172,6 +172,7 @@ func (g *Generic) Run(ctx context.Context) error {
 	// Stage: start all informers then wait on them
 	stage := stgr.NextStage()
 	for _, inf := range g.Informers {
+		inf := inf // capture field into a scoped variable to avoid data race
```

Comment:
stage.StartWithChannel(inf.Run) -> what's below in a previous PR. I think it occurs now because the func closes over a top level reference compared to passing a function pointer each loop iteration. (different pointer each time)